core/src/main/java/com/google/instrumentation/trace/TraceService.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.trace;
+
+/**
+ * Class that holds the implementation instances for {@link Tracer} and {@link
+ * BinaryPropagationHandler}.
+ *
+ * <p>This implementation of this class is loaded using reflection by the {@link Trace}.
+ */
+public abstract class TraceService {
```

Comment:
So one reason to have a separate class that holds the instances instead of having Tracer keeping it's own instance and propagation handler doing the same is because we try to have only one class that keeps the "global state".

Another reason is that we want to add more functionality to this class like:
* let users register "exporters"
* etc.
that in my opinion does not belong to tracer (or it may belong but then Tracer become a TraceService which includes multiple different functionalities).

In the current desing Tracer deals with Span creation and propagation in the process which are the most used functionalities.